packages/url_launcher/url_launcher/android/src/main/java/dev/flutter/plugins/urllauncher/UrlLauncher.java

```diff
@@ -0,0 +1,87 @@
+package dev.flutter.plugins.urllauncher;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Browser;
+import android.support.annotation.Nullable;
+
+/** Launches components for URLs. */
+class UrlLauncher {
+  private Context activityContext;
+
+  /**
+   * Uses the given {@code activityContext} for launching intents.
+   *
+   * <p>It may be null initially, but should be set before calling {@link #launch}.
+   */
+  UrlLauncher(@Nullable Context activityContext) {
```

Comment:
If you really need an `Activity` `Context` then I would recommend only accepting an `Activity`. It will enforce your requirements with the type system and avoid any future developer mistakes by overlooking the significance of `activityContext` vs `context`.

Reply:
Done. I took a second look at the original code, and I oversimplified. Half the time it was using an activity and half the time it was using application context. Split it out into separate objects and kept the `Activity` as that class following this suggestion.